src/behaviors/filterable-store.js

```diff
@@ -0,0 +1,146 @@
+import angular from 'angular';
+import {Behavior} from './behavior';
+import {Handler as handlerDecorator} from '../store';
+
+import {
+  Transformer,
+  Transformable as transformableDecorator
+} from './transformable';
+
+import {
+  addBehavior,
+  camelcase,
+  Inject as injectDecorator
+} from '../utils';
+
+export class FilterableStoreBehavior extends Behavior {
+  constructor(instance, {collection} = {}) {
+    super(...arguments);
+
+    this.collection = collection || 'items';
+    this.transformerWeight = 25;
+    this.filters = new Map();
+  }
+
+  get $filter() {
+    return this.instance.$filter;
+  }
+
+  get transformableCollection() {
+    return this.instance.transformables[this.collection];
+  }
+
+  get transformer() {
+    if (!this._transformer) {
+      this._transformer = new Transformer('filterableStore', items => items, this.transformerWeight);
+    }
+    return this._transformer;
+  }
+
+  onFilterChange(filterName, filter) {
+    if (Reflect.apply(Object.prototype.toString, filter) === '[object String]') {
+      this.filters.set(filterName, items => {
+        return this.$filter('filter')(items, filter);
+      });
+    } else {
+      this.filters.set(filterName, items => {
```

Comment:
Is this similar to the logic of the current filterable store? I wonder if this is all we need but I guess in connect we havent found any other usecases